packages/testing/src/TestStack.ts

```diff
@@ -0,0 +1,83 @@
+import { tmpdir } from 'node:os';
+import { join } from 'node:path';
+import { readFile } from 'node:fs/promises';
+import { App, Stack } from 'aws-cdk-lib';
+import { AwsCdkCli, RequireApproval } from '@aws-cdk/cli-lib-alpha';
+import type { ICloudAssemblyDirectoryProducer } from '@aws-cdk/cli-lib-alpha';
+
+/**
+ * Test stack that can be deployed to the selected environment.
+ */
+class TestStack implements ICloudAssemblyDirectoryProducer {
+  /**
+   * Reference to the AWS CDK App object.
+   * @default new App()
+   */
+  public appRef: App;
+  /**
+   * Reference to the AWS CDK Stack object.
+   * @default new Stack(this.appRef, stackName)
+   */
+  public stackRef: Stack;
+  /**
+   * @internal
+   * Reference to the AWS CDK CLI object.
+   */
+  #cli: AwsCdkCli;
+
+  public constructor(stackName: string, app?: App, stack?: Stack) {
+    this.appRef = app ?? new App();
+    this.stackRef = stack ?? new Stack(this.appRef, stackName);
+    this.#cli = AwsCdkCli.fromCloudAssemblyDirectoryProducer(this);
+  }
+
+  /**
+   * Deploy the test stack to the selected environment.
+   *
+   * It returns the outputs of the deployed stack.
+   */
+  public async deploy(): Promise<Record<string, string>> {
+    const outputFilePath = join(
+      tmpdir(),
+      'powertools-e2e-testing',
+      `${this.stackRef.stackName}.outputs.json`
+    );
+    await this.#cli.deploy({
+      stacks: [this.stackRef.stackName],
+      requireApproval: RequireApproval.NEVER,
+      outputsFile: outputFilePath,
+    });
+
+    return JSON.parse(await readFile(outputFilePath, 'utf-8'))[
+      this.stackRef.stackName
+    ];
+  }
+
+  /**
+   * Destroy the test stack.
+   */
+  public async destroy(): Promise<void> {
+    await this.#cli.destroy({
+      stacks: [this.stackRef.stackName],
+      requireApproval: false,
```

Comment:
This is odd, the CLI command `destroy` doesn't have `require-approval` but rather `force` option. You can even see that the type of the same field is different between `deploy` and `destroy`. I will submit an issue in the cdk repo.

Reply:
It still works, though the CDK team remap the parameter to `--force`

https://github.com/aws/aws-cdk/blob/b1ca3c09e68a2c1f5bf5ce4c9c40f12db7f1767f/packages/%40aws-cdk/cli-lib-alpha/lib/cli.ts#L233C10-L233C33

Reply:
Agree that we should submit an issue (& maybe PR). I was also confused by the naming, probably they it's a mistake.